applications/ChimeraApplication/custom_processes/rotate_region_process.cpp

```diff
@@ -0,0 +1,357 @@
+    return;
+
+  mTime = time;
+  // Does the time integration and calculates the new theta and omega
+  CalculateCurrentRotationState();
+
+  const int num_nodes = static_cast<int>(mrModelPart.NumberOfNodes());
+  const NodeIteratorType it_node_begin = mrModelPart.NodesBegin();
+
+#pragma omp parallel for schedule(guided, 512)
+  for (int i_node = 0; i_node < num_nodes; ++i_node)
+  {
+    auto it_node = it_node_begin;
+    std::advance(it_node, i_node);
+
+    /// Calculating the displacement of the current node
+    array_1d<double, 3> transformed_coordinates;
+    TransformNode(it_node->GetInitialPosition().Coordinates(),
+                  transformed_coordinates, mTheta);
+
+    it_node->X() = transformed_coordinates[0];
+    it_node->Y() = transformed_coordinates[1];
+    if (domain_size > 2)
+      it_node->Z() = transformed_coordinates[2];
+
+    it_node->FastGetSolutionStepValue(ROTATION_MESH_DISPLACEMENT) = transformed_coordinates - it_node->GetInitialPosition().Coordinates();
+
+    // Computing the linear velocity at this it_node
+    DenseVector<double> radius(3);
+    DenseVector<double> linear_velocity(3);
+    radius[0] = it_node->X() - mCenterOfRotation[0];
+    radius[1] = it_node->Y() - mCenterOfRotation[1];
+    radius[2] = it_node->Z() - mCenterOfRotation[2];
+    CalculateLinearVelocity(mAxisOfRotationVector, radius, linear_velocity);
+    it_node->FastGetSolutionStepValue(ROTATION_MESH_VELOCITY_X, 0) =
+        mAngularVelocityRadians * linear_velocity[0];
+    it_node->FastGetSolutionStepValue(ROTATION_MESH_VELOCITY_Y, 0) =
+        mAngularVelocityRadians * linear_velocity[1];
+    if (domain_size > 2)
+      it_node->FastGetSolutionStepValue(ROTATION_MESH_VELOCITY_Z, 0) =
+          mAngularVelocityRadians * linear_velocity[2];
+
+    if (mParameters["is_ale"].GetBool())
+    {
+      it_node->FastGetSolutionStepValue(MESH_VELOCITY_X, 0) =
+            it_node->FastGetSolutionStepValue(ROTATION_MESH_VELOCITY_X, 0);
+      it_node->FastGetSolutionStepValue(MESH_VELOCITY_Y, 0) =
+            it_node->FastGetSolutionStepValue(ROTATION_MESH_VELOCITY_Y, 0);
+      if (domain_size > 2)
+        it_node->FastGetSolutionStepValue(MESH_VELOCITY_Z, 0) =
+            it_node->FastGetSolutionStepValue(ROTATION_MESH_VELOCITY_Z, 0);
+
+      if (it_node->IsFixed(VELOCITY_X))
+        it_node->FastGetSolutionStepValue(VELOCITY_X, 0) =
+            it_node->FastGetSolutionStepValue(MESH_VELOCITY_X, 0);
+
+      if (it_node->IsFixed(VELOCITY_Y))
+        it_node->FastGetSolutionStepValue(VELOCITY_Y, 0) =
+            it_node->FastGetSolutionStepValue(MESH_VELOCITY_Y, 0);
+
+      if (domain_size > 2)
+        if (it_node->IsFixed(VELOCITY_Z))
+          it_node->FastGetSolutionStepValue(VELOCITY_Z, 0) =
+              it_node->FastGetSolutionStepValue(MESH_VELOCITY_Z, 0);
+    }
+  }
+
+  KRATOS_CATCH("");
+}
+
+void RotateRegionProcess::ExecuteFinalizeSolutionStep() {}
+
+void RotateRegionProcess::ExecuteAfterOutputStep() {}
```

Comment:
If you want to do nothing you can leave the implementation in the base `Process` class.